MultisigTests/UI/Safe Management/Create Safe/Web3AuthMFAServiceTest.swift

```diff
@@ -0,0 +1,172 @@
+//
+//  Web3AuthMFAServiceTest.swift
+//  MultisigTests
+//
+//  Created by Dirk Jäckel on 09.08.23.
+//  Copyright © 2023 Gnosis Ltd. All rights reserved.
+//
+
+import XCTest
+import tkey_pkg
+@testable import Multisig
+
+class Web3AuthMFAServiceTests: XCTestCase {
+
+    var web2authMFAService: Web3AuthMFAService!
+    var keychain: TestKeychainInterface!
+    let aDeviceShare = "66665182a2b416cfe448413513dfc6ee8d87538b8d678bde30177c1f463aa822"
+    let passwordShare = "4b5a18ed39e4b369fdcda96bdeeb7ec7375b3cdb89b5fb37de8635c8e07382d7"
+    let postBoxKey = "c57c57f1a3463f14fb6ce79835f5df8437a8d449b5e2219aa2bb3876554f99cb"
+    let publicAddress = "0x875b5EAAC06a857d1046cdA1b2a6683deeFbA5B4"
+    let password = "foobar23"
+    let finalKey = "75907aad04675827696b92ca481c6b00a24514d8b8819c97840c1960a50f7126"
+
+    override func setUp() async throws {
+        keychain = TestKeychainInterface()
+    }
+
+    override func tearDown() {
+    }
+
+    func testInitWithDeviceShare() async throws {
+        // init with device share
+        keychain.dict["\(publicAddress):device-key"] = aDeviceShare
+        try await web2authMFAService = Web3AuthMFAService(postBoxKey: postBoxKey,
+                                                          publicAddress: publicAddress,
+                                                          keychainInterface: keychain)
+        try await web2authMFAService.reconstruct()
```

Comment:
do I understand right that this might take some time, and it depends on the network? If it does, then these tests should be included in the integration tests target and not in the unit tests. 

Otherwise, this will slow down or make the unit tests flaky.